### DISCUSSION TITLE
Open url by clicking a button

### DISCUSSION THREAD

**Discussion by Zerachiel01** [unoplatform/uno#17354] [Q&A] [answered]
Hello everyone!

As the title implies, I want to add a button to my app that opens a url. Normally I would use the HyperlinkButton-Control, but I don't like its visual appearance (it does not look like a button). I tried making it look like a button by putting it in a Border-Control, unfortunately this does not make the whole Border clickable and I am not able to change the HyperlinkButton's appearance, since changing its Foreground-Property does not change the text color (Issue?).
Then I tried using a normal Button-Control, but I could not find a proper method to open a url for a uno platform app. Maybe someone else knows what to call? The HyperlinkButton-Control seems to know such a method but I did not find one.

Thanks in advance!

**Comment by morning4coffe-dev** [comment, marked as answer]
Hey @Zerachiel01,

for opening a link with a regular Button, you can use the [URI Launcher](https://platform.uno/docs/articles/features/windows-system.html) from Windows.System and use Click or bind it to a Command, example: 
```xml
<Button Click="LaunchWebsiteButtonAsync_Click" Content="Launch" />
```

```csharp
private async void LaunchWebsiteButtonAsync_Click(object sender, RoutedEventArgs e)
{
    if(Uri.TryCreate(LaunchUriText, UriKind.Absolute, out var uri))
	{
		await Launcher.LaunchUriAsync(uri);
	}
	
	//error
}
```

**Reply by Zerachiel01** [reply]
Thanks a lot, that works for me.I don't know why I couldn't find the URI Laucher. I think I wasn't sure if a class from a _Windows_.System namespace would work on android.

**Reply by morning4coffe-dev** [reply]
No problem, glad I could help! It does work on Android and other platforms too, even with support for nested URIs, `QueryUriSupportAsync`, etc.

You can also check our overview of supported namespaces here: https://platform.uno/docs/articles/using-uno-ui.html#winrt-features-non-visual-apis
